src/posix/platform/udp.cpp

```diff
@@ -402,19 +436,22 @@ void platformUdpInit(const char *aIfName)
         DieNow(OT_EXIT_INVALID_ARGUMENTS);
     }
 
-    sPlatNetifIndex = if_nametoindex(aIfName);
-
-    if (sPlatNetifIndex == 0)
+    if (aIfName != gNetifName)
```

Comment:
nit:
```suggestion
    if (strcmp(aIfName, gNetifName))
```

Reply:
This is intended, PLATFORM_NETIF pass gNetifName to to PLATFORM_UDP to initialize it.